src/components/Card/Card.tsx

```diff
@@ -5,74 +5,89 @@ import {
   useMultiStyleConfig,
   chakra,
   HStack,
-  Text,
-  isStyleProp,
   StackProps,
+  StylesProvider,
+  useStyles,
+  omitThemingProps,
 } from "@chakra-ui/react"
-import { objectFilter } from "@chakra-ui/utils"
 import { ButtonProps } from "@opengovsg/design-system-react"
-import React from "react"
+import { createContext, useContext } from "react"
 
 import { CardVariant, CARD_THEME_KEY } from "theme/components/Card"
 
 interface CardProps
   extends Omit<ButtonProps, "leftIcon" | "rightIcon" | "variant"> {
-  icon?: JSX.Element
-  actionButton?: JSX.Element
   // eslint-disable-next-line react/require-default-props
   variant?: CardVariant
 }
 
-export const Card = (props: BoxProps): JSX.Element => {
-  const styles = useMultiStyleConfig(CARD_THEME_KEY, props)
-  return <Box position="relative" __css={styles.container} {...props} />
+interface CardContextProps extends ReturnType<typeof omitThemingProps> {
+  variant: CardVariant
 }
 
-export const CardContent = ({
-  icon,
-  children,
-  actionButton,
-  ...props
-}: CardProps): JSX.Element => {
-  const styleProps = objectFilter(props, (_, prop) => isStyleProp(prop))
-  const remainingProps = objectFilter(props, (_, prop) => !isStyleProp(prop))
+const CardContext = createContext<CardContextProps | undefined>(undefined)
 
-  const styles = useMultiStyleConfig(CARD_THEME_KEY, props)
-  const { variant } = props
-  const shouldWrap =
-    React.Children.count(children) === 1 && typeof children === "string"
-  const wrappedChildren = shouldWrap ? (
-    <Text textStyle="body-1">{children}</Text>
-  ) : (
-    children
-  )
+const useCardContext = (): CardContextProps => {
+  const context = useContext(CardContext)
+  if (!context) {
+    throw new Error("useCalendar must be used within a CardProvider")
+  }
 
-  return <Box position="relative">{children}</Box>
+  return context
 }
 
-export const CardBody = (props: StackProps) => {
+export const Card = ({ children, ...props }: CardProps): JSX.Element => {
+  const variant = props.variant ?? "single"
   const styles = useMultiStyleConfig(CARD_THEME_KEY, props)
+  const buttonBehaviourProps = omitThemingProps(props)
+  const Inner = () =>
+    variant === "multi" ? (
+      <chakra.button __css={styles.container} {...props}>
+        {children}
+      </chakra.button>
```

Comment:
Just for info, why do we use chakra.button here?

Reply:
there's actually 3 kinds of buttons!
 `<Button />` is a component that comes with the styling, so it'll have focus/hover states etc together with padding and what not 

The one above is just a basic HTML button that hooks into chakra, so passing colour tokens work and so does chakra shorthand (p for padding etc). This is also what chakra's `Button` is implemented with 

The last is just `button`, which is basically the above without chakra stuff.

I'm using `chakra.button` here because I want the __behaviour__ but not the styling as the styling should be card specific